packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -279,6 +299,27 @@ export default util.createRule<Options, MessageIds>({
           return;
         }
 
+        if (isUnionOrIntersectionType(type)) {
+          if (
+            ignorePrimitives?.string &&
+            type.types.some(t => isTypeFlagSet(t, ts.TypeFlags.String))
+          ) {
+            return;
+          }
+          if (
+            ignorePrimitives?.boolean &&
+            type.types.some(t => isTypeFlagSet(t, ts.TypeFlags.BooleanLiteral))
+          ) {
+            return;
+          }
+          if (
+            ignorePrimitives?.number &&
+            type.types.some(t => isTypeFlagSet(t, ts.TypeFlags.Number))
```

Comment:
[Request] There are three `.some `loops here. You can reduce the amount of loops and lines of code a teeny bit by using `|` on all the ignorable type flags.

```ts
const ignorableFlags [
  ignorePrimitives.boolean && ts.TypeFlags.BooleanLiteral,
  ignorePrimitives.number && ts.TypeFlags.Number,
  ignorePrimitives.string && ts.TypeFlags.String,
]
  .filter(isNotUndefined)
  .reduce((previous, flag) => previous | flag, 0);
```